diff -ruN code2html-0.9.1/code2html code2html-0.9.1.new/code2html
--- code2html-0.9.1/code2html	2002-01-12 22:17:02.000000000 +0100
+++ code2html-0.9.1.new/code2html	2012-03-03 19:28:46.000000000 +0100
@@ -2258,7 +2258,7 @@
 # taken from nedit
 # modified by PP
 $LANGUAGE{'c++'}        = {
-                            'filename'   => '\\.(c(c|pp|xx)|h(h|pp|xx)|C(C|PP|XX)?|H(H|PP|XX)?|i)$',
+                            'filename'   => '\\.(c(c|pp|xx)|h(h|pp|xx)|C(C|PP|XX)?|H(H|PP|XX)?|i|qml)$',
                             'regex'      => '',
                             'patterns'   => [
                                               {
